Sources/iOSClientExposure/Extensions/UIDevice+Extensions.swift

```diff
@@ -22,4 +22,80 @@ extension UIDevice {
         if systemName == "iPhone OS" { return "iOS" }
         return systemName
     }
+    
+    var modelName: String {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children.reduce("") { identifier, element in
+            guard let value = element.value as? Int8, value != 0 else { return identifier }
+            return identifier + String(UnicodeScalar(UInt8(value)))
+        }
+        if let path = Bundle.main.path(forResource: "AppleTVModels", ofType: "plist"),
+           let dict = NSDictionary(contentsOfFile: path) as? [String: Any],
+           let name = dict[identifier] as? String {
+            return name
+        }
+        
+        return identifier
+    }
+    
+    var appleTVModel: String {
+        var systemInfo = utsname()
+        uname(&systemInfo)
+        let machineMirror = Mirror(reflecting: systemInfo.machine)
+        let identifier = machineMirror.children.reduce("") { identifier, element in
+            guard let value = element.value as? Int8, value != 0 else { return identifier }
+            return identifier + String(UnicodeScalar(UInt8(value)))
+        }
+        switch identifier {
+        case "AppleTV1,1": return "atv-1"
+        case "AppleTV2,1": return "atv-2"
+        case "AppleTV3,1", "AppleTV3,2", "AppleTV3,3": return "atv-3"
+        case "AppleTV5,3": return "atv-4"
+        case "AppleTV6,2": return "atv-4k-1"
+        case "AppleTV11,1": return "atv-4k-2"
+        case "AppleTV11,2", "AppleTV11,3": return "atv-4k-3"
+        default: return "atv-unknown"
```

Comment:
Seems to me that you could make an algorithm around this instead.
basically lowercase the string and then split it on the first digit and replace `,` with `-`.
So the deviceModel becomes `appletv-11-1` the mapping to 4k isn't necessary imho .

Reply:
I just followed the format of the confluence documentation , if we all agree to the pass this format `appletv-11-1` I can fix it.  I agree hardcoding values are not the optimal in this case & would be nice to pass a format that we dynamically create.

Reply:
Updated the PR :)